plugins/node/opentelemetry-instrumentation-bunyan/src/types.ts

```diff
@@ -21,5 +21,24 @@ import { InstrumentationConfig } from '@opentelemetry/instrumentation';
 export type LogHookFunction = (span: Span, record: Record<string, any>) => void;
 
 export interface BunyanInstrumentationConfig extends InstrumentationConfig {
+  /**
+   * Whether to disable the automatic sending of log records to the
+   * OpenTelemetry Logs Bridge API.
+   * @default false
+   */
+  disableLogsBridge?: boolean;
+
+  /**
+   * Whether to disable the injection of data such as trace-context fields into
+   * log output.
+   * @default false
+   */
+  disableInjection?: boolean;
```

Comment:
This config is causing me a little bit of confusion, enable in the actual InstrumentionConfig serve the same purpose, I know the instrumentation in this case serves multiple functionality now, maybe updating the name to disableTraceContextInjection would be better, also it would be good to define a common interface for Log Instrumentations so all following will follow the same structure.

Reply:
> enable in the actual InstrumentionConfig serve the same purpose

Yes, having two enable/disable booleans in addition to the `InstrumentationConfig.enabled` feels unfortunate. In Java-land there are separate packages for (a) trace-context injection and (b) the logs bridging. However, we already have an instrumentation just called "instrumentation-bunyan", so it would be a little awkward to have two separate packages, one called "instrumentation-bunyan" for the former and a separate one for the logs bridging.


> maybe updating the name to disableTraceContextInjection would be better

I'm okay with that name. Note that `disableInjection` also controls whether the optional `logHook` function is called to support "injecting" other attributes. However, the primary use is for trace-context.

Another possible name could be something like `disableLogCorrelation`. "Log Correlation" is language that is used at https://opentelemetry.io/docs/specs/otel/logs/#log-correlation and https://opentelemetry-python-contrib.readthedocs.io/en/latest/instrumentation/logging/logging.html#envvar-OTEL_PYTHON_LOG_CORRELATION

Thoughts?



> also it would be good to define a common interface for Log Instrumentations so all following will follow the same structure.

My hope/intent is to follow up with similar PRs for Pino and Winston. I plan to use the same language/structure for configuration and docs there. Do you think it merits/requires a shared `LoggingInstrumentationConfig` somewhere? (Perhaps in `@opentelemetry/instrumentation`? Adding a new package just for this shared type feels like overkill.)

Reply:
disableLogCorrelation sounds way better to me, yeah having a LoggingInstrumentationConfig and a LoggingInstrumentation in @opentelemetry/instrumentation starts to make sense to me, at least for now we want to have, Winston, Bunyan, Pino and Console to also generate log telemetry, so this is something that several package would benefit from.

Reply:
> disableLogCorrelation sounds way better to me

commit c740b767 for this